src/messages.cpp

```diff
@@ -343,14 +357,20 @@ void Messages::display_messages(WINDOW *const ipk_target, int const left, int co
             }
 
             const nc_color col = m.get_color(player_messages.impl_->curmes);
-            const auto folded_strings = foldstring(m.get_with_count(), maxlength);
+            std::string message_text = m.get_with_count();
+            if (!m.is_recent(player_messages.impl_->curmes)) {
+                message_text = remove_color_tags(message_text);
+            }
+
+            const auto folded_strings = foldstring(message_text, maxlength);
             const auto folded_rend = folded_strings.rend();
             for( auto string_iter = folded_strings.rbegin();
                     string_iter != folded_rend && line >= top; ++string_iter, line-- ) {
                 // Redrawing line to ensure new messages similar to previous
                 // messages will not be missed by screen readers
                 wredrawln(ipk_target, line, 1);
-                mvwprintz(ipk_target, line, left, col, "%s", string_iter->c_str());
+                nc_color col_out = col;
+                print_colored_text( ipk_target, line, left, col_out, col, string_iter->c_str());
```

Comment:
`print_colored_text` does not do formating, but it accepts a `std::string` as text, not a `const char*`. The `->c_str()` is useless, use `*string_iter` to forward the string object directly.